replicatelogs.go

```diff
@@ -677,3 +689,42 @@ func readTenantMassifChanges(ctx context.Context, cCtx *cli.Context, cmd *CmdCtx
 	// No explicit config and --all not set, read from stdin
 	return stdinToDecodedTenantMassifs()
 }
+
+// prefetchingSealReader pre-fetches the seal for the massif to avoid racing with the
+// sealer.  If the massif is read first, the log can grow and a a new seal can
+// be applied to the *longer* log. At which point the previously read copy of
+// the massif will be "to short" for the seal.
+// See Bug#10530
+type prefetchingSealReader struct {
+	msg            *cose.CoseSign1Message
+	state          massifs.MMRState
+	tenantIdentity string
+	massifIndex    uint32
+}
+
+var ErrInconsistentUseOfPrefetchedSeal = errors.New("prefetching signed root reader used inconsistently")
```

Comment:
can we define types at the top of the file?

Reply:
as well as vars :)

Reply:
yep, done